synapse/lib/hive.py

```diff
@@ -860,11 +864,22 @@ class HiveUser(HiveIden):
         prof = await self.node.open(('profile',))
         self.profile = await prof.dict()
 
+        # vars cache for persistent user level data storage
+        # TODO: max size check?
+        pvars = await self.node.open(('vars',))
+        self.pvars = await pvars.dict()
+
         self.fullrules = []
         self.permcache = s_cache.FixedCache(self._calcPermAllow)
 
         self._initFullRules()
 
+        # user needs to get hive:get / hive:set / hive:pop permissions to their stormvars tree
+        for baseperm in ('hive:get', 'hive:set', 'hive:pop'):
```

Comment:
Technically speaking, a rule could be in an order that still caused it not to work even though it's present in the set.  In the case of the system doing work on behalf of the user, it's ok avoid permissions checks under the assumption that the system already knows it's that user.  Adding perms tuples here could get confusing when showing user rules.  Lets act on the users behalf under the assumption that we ( the storm runtime ) are controlling access.